pkg/process/config/yaml_config.go

```diff
@@ -353,8 +353,11 @@ func (a *AgentConfig) LoadProcessYamlConfig(path string) error {
 	// Used to override container source auto-detection
 	// and to enable multiple collector sources if needed.
 	// "docker", "ecs_fargate", "kubelet", "kubelet docker", etc.
-	if sources := config.Datadog.GetStringSlice(key(ns, "container_source")); len(sources) > 0 {
-		util.SetContainerSources(sources)
+	containerSourceKey := key(ns, "container_source")
+	if config.Datadog.Get(containerSourceKey) != nil {
+		if sources := config.Datadog.GetStringSlice(containerSourceKey); len(sources) > 0 {
+			util.SetContainerSources(sources)
+		}
```

Comment:
why do you have to call `Get` *and* `GetStringSlice` ?

Reply:
`Get` will not log a warning if the config is nil

Reply:
if not nil I rely on `GetStringSlice` to cast it, I guess that better than manually casting it

Reply:
currently the if `DD_PROCESS_AGENT_CONTAINER_SOURCE` is not set (which is the case most of the time) the process agent (and the cluster agent when the orch explorer is enabled) always log that warning when they start.
happy to consider other alternatives

Reply:
@leeavital

Reply:
This seems fine to me if it's a spammy log, but we might consider writing a wrapper or something around `GetStringSlice` with a different log level or no log. I think in most (all?) other places in the code we let it log.

Reply:
no it doesn't happen in all places because we set default values in general (empty string slice), but for this config in particular we don't set a default value, I don't know why TBH, the processes team might have more context

Reply:
Right, makes sense. In that case LGTM, would be curious about why there's no default for this though.

Reply:
seems like all process configs are just `SetKnown` https://github.com/DataDog/datadog-agent/blob/master/pkg/config/config.go#L662

Reply:
In that case, I think we should either add a comment explaining as much, or fix GetStringSlice to not log.  It's not super clear why the code is written this way just from looking at it.

Reply:
will add a comment, I believe `GetStringSlice` is behaving as expected it's just the process agent config fields don't have default values

Reply:
added a comment, let me know if it's clearer now